pandas/io/pytables.py

```diff
@@ -313,10 +313,14 @@ def read_hdf(path_or_buf, key=None, **kwargs):
 
         """
 
-    if kwargs.get('mode', 'a') not in ['r', 'r+', 'a']:
+    # Ignore any supplied mode, we only need to read data
```

Comment:
cancel that, there is already code to check for existence and raise FileNotFoundError